### PR TITLE
Feature/user and comment crud

### PR DESCRIPTION
**Pull request by MaxGudinovich**
Added user and comment CRUD
Created new table for saving password

### CODE REVIEW

File: News/Configurations/CommentConfiguration.cs [hunk 1]
```diff
@@ -9,5 +9,8 @@ internal class CommentConfiguration : IEntityTypeConfiguration<CommentEntity>
     public void Configure(EntityTypeBuilder<CommentEntity> builder)
     {
```
Comment (AvaNice) [inline News/Configurations/CommentConfiguration.cs:10]: Hey mate! Add toTable pls

File: News/Configurations/PasswordConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using News.Entities;
+
+namespace News.Configurations;
+
+public class PasswordConfiguration : IEntityTypeConfiguration<PasswordEntity>
+{
+    public void Configure(EntityTypeBuilder<PasswordEntity> builder)
+    {
```
Comment (AvaNice) [inline News/Configurations/PasswordConfiguration.cs:10]: Same here. Pls add to table just to make it clear

File: News/Configurations/PublisherConfiguration.cs [hunk 3]
```diff
@@ -12,6 +12,7 @@ public PublisherConfiguration()
 
     public void Configure(EntityTypeBuilder<PublisherEntity> builder)
     {
+        //toTable
```
Comment (AvaNice) [inline News/Configurations/PublisherConfiguration.cs]: TODO:

File: News/Controllers/CommentsController.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    public CommentsController(ICommentService service)
+    {
+        _service = service;
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> GetCommentById([FromQuery] Guid id)
+    {
+        var comment = await _service.GetCommentById(id);
+        return Ok(comment);
```
Comment (AvaNice) [inline News/Controllers/CommentsController.cs:23]: pls add blank line before return

File: News/Controllers/CommentsController.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+{
+    private readonly ICommentService _service;
+
+    public CommentsController(ICommentService service)
+    {
+        _service = service;
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> GetCommentById([FromQuery] Guid id)
```
Comment (AvaNice) [inline News/Controllers/CommentsController.cs]: take a look at possibility to make id from route

File: News/Controllers/UsersController.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+{
+    private readonly IUserService _service;
+
+    public UsersController(IUserService service)
+    {
+        _service = service;
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> GetUserById([FromQuery] Guid id)
```
Comment (AvaNice) [inline News/Controllers/UsersController.cs]: same as prev controller

File: News/DataAccess/NewsDb.cs [hunk 7]
```diff
@@ -8,7 +8,7 @@ public class NewsDb : DbContext
 {
     public DbSet<UserEntity> Users { get; set; }
 
-    public DbSet<CommentEntity> Comments { get; set; }
+    public DbSet<UserEntity> Comments { get; set; }
```
Comment (AvaNice) [inline News/DataAccess/NewsDb.cs]: dont think dat should be UserEntity. Mb string will be better

File: News/DataAccess/Repo/CommentRepo.cs [hunk 8]
```diff
@@ -14,4 +14,36 @@ public CommentRepo(DbContext context)
...
+        //Include
+        var entity = (await _dbSet.AddAsync(comment)).Entity;
+        await _context.SaveChangesAsync();
+
+        return entity;
+    }
+
+    public async Task<CommentEntity> UpdateComment(CommentEntity comment)
+    {
+        var entity = _dbSet.Update(comment).Entity;
```
Comment (AvaNice) [inline News/DataAccess/Repo/CommentRepo.cs:35]: does ef have updateAsync ?

File: News/Entities/Models/CommentModel.cs [hunk 9]
```diff
@@ -10,3 +12,5 @@ public class CommentModel
 
     public int LikeCount { get; set; }
 }
+//should add created at and updated at? 
+//rename comment to content
```
Comment (AvaNice) [inline News/Entities/Models/CommentModel.cs:16]: TODO:

File: News/Entities/Models/CommentModel.cs [hunk 10]
```diff
@@ -2,6 +2,8 @@
 
 public class CommentModel
 {
+    public Guid Id { get; set; }
+
     public string Comment { get; set; }
```
Comment (AvaNice) [inline News/Entities/Models/CommentModel.cs:7]: Comment => content? or it's out of scope of this PR?

File: News/Entities/Models/CommentResponseModel.cs [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+    public string Comment { get; set; }
+
+    public Guid UserId { get; set; }
+
+    public Guid ArticleId { get; set; }
+
+    public int LikeCount { get; set; }
+
+    //without User only string Username
```
Comment (AvaNice) [inline News/Entities/Models/CommentResponseModel.cs]: if that line related to Username property we dont' need blank line betwen them

File: News/Entities/Models/CommentResponseModel.cs [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+    public string Username { get; set; }
+    public UserResponseModel User { get; set; }
```
Comment (AvaNice) [inline News/Entities/Models/CommentResponseModel.cs:17]: add blank line between properties

File: News/Mapping/NewsMappingProfile.cs [hunk 13]
```diff
@@ -9,16 +9,20 @@ public class NewsMappingProfile : Profile
...
         CreateMap<TagModel, TagEntity>().ReverseMap();
         CreateMap<ArticleTagModel, ArticleTagEntity>().ReverseMap();
         CreateMap<PublisherModel, PublisherEntity>().ReverseMap();
-
-        CreateMap<UserCreateModel, UserEntity>()
-            .ForMember(dest => dest.Password, act => act.Ignore())
-            .ForMember(dest => dest.Id, opt => opt.MapFrom<UserEntityIdResolver>())
-            .ForMember(dest => dest.Salt, opt => opt.MapFrom<UserEntityPasswordResolver>());
+        CreateMap<UserResponseModel, UserEntity>().ReverseMap();
+        //CreateMap<UserCreateModel, UserEntity>()
```
Comment (AvaNice) [inline News/Mapping/NewsMappingProfile.cs]: commented code

File: News/Mapping/Resolvers/CommentEntityIdResolver.cs [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+using News.Entities;
+using AutoMapper;
+
+namespace News.Mapping.Resolvers;
+
+internal class CommentEntityIdResolver : IValueResolver<CommentCreateModel, CommentEntity, Guid>
+{
+    public Guid Resolve(CommentCreateModel source, CommentEntity destination, Guid destMember, ResolutionContext context)
+    {
+        return Guid.NewGuid();
```
Comment (AvaNice) [inline News/Mapping/Resolvers/CommentEntityIdResolver.cs]: we don't need to create resolver for such simple action. use mapFrom(_ => new Guid

File: News/Migrations/20240928083245_AddCommentEntityTable.cs [hunk 15]
```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace News.Migrations;
+
+/// <inheritdoc />
+public partial class AddCommentEntityTable : Migration
```
Comment (AvaNice) [inline News/Migrations/20240928083245_AddCommentEntityTable.cs:8]: do not commit migrations

Comment (AvaNice) [reply News/Migrations/20240928083245_AddCommentEntityTable.cs:8]: .gitignore

File: News/Migrations/20240929092228_AddPasswordInfoTable.cs [hunk 16]
```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace News.Migrations;
+
+/// <inheritdoc />
+public partial class AddPasswordInfoTable : Migration
```
Comment (AvaNice) [inline News/Migrations/20240929092228_AddPasswordInfoTable.cs:9]: same

**Review by AvaNice — CHANGES_REQUESTED** [review]
Good job! Few minor comments should be resolved but it's steel a good job

File: News/DataAccess/Repo/CommentRepo.cs (hunk 8 above)
Comment (MaxGudinovich) [reply News/DataAccess/Repo/CommentRepo.cs:35]: EF doesn't have it

File: News/Entities/Models/CommentModel.cs (hunk 9 above)
Comment (MaxGudinovich) [reply News/Entities/Models/CommentModel.cs:16]: Where can I save my TODO`s if that out of scope now?

File: News/Entities/Models/CommentModel.cs (hunk 10 above)
Comment (MaxGudinovich) [reply News/Entities/Models/CommentModel.cs:7]: Will be updated soon. I have a few questions about it.

File: News/Configurations/CommentConfiguration.cs (hunk 1 above)
Comment (AvaNice) [reply News/Configurations/CommentConfiguration.cs:10]: ToTable was added :)

**Pull request merged**